x/wasm/internal/keeper/ioutil_test.go

```diff
@@ -0,0 +1,63 @@
+package keeper
+
+import (
+	"errors"
+	"io"
+	"io/ioutil"
+	"strings"
+	"testing"
+
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+)
+
+func TestUncompress(t *testing.T) {
+	wasmRaw, err := ioutil.ReadFile("./testdata/contract.wasm")
+	require.NoError(t, err)
+
+	wasmGzipped, err := ioutil.ReadFile("./testdata/contract.wasm.gzip")
+	require.NoError(t, err)
+
+	specs := map[string]struct {
+		src       []byte
+		expError  error
+		expResult []byte
+	}{
+		"handle wasm uncompressed": {
+			src:       wasmRaw,
+			expResult: wasmRaw,
+		},
+		"handle wasm compressed": {
```

Comment:
Can you also add an invalid zip file?
I think `wasmGzipped[:len(wasmGzipped)-5]` should produce an error (as there is a checksum).

Reply:
Besides that (and the zip bomb), nice test coverage

Reply:
Added your case. I could not make a gzip bomb work with the code so I have added a test that checks a regular file with content > max.
It works all nicely with the `LimitedReader`. The only problem is that the error message does not reflect the `ContractLimitExceeded` constraint but is `ErrUnexpectedEOF`.
Checking `LimitedReader.N <=0` did not work as `N` was not updated for some reasons